lib/routes/routeBackbeat.js

```diff
@@ -472,22 +472,61 @@ function putMetadata(request, response, bucketInfo, objMd, log, callback) {
                 omVal[headerName] = objMd[headerName];
             });
         }
-        const versionId = decodeVersionId(request.query);
-        // specify both 'versioning' and 'versionId' to create a "new"
-        // version (updating master as well) but with specified
-        // versionId
-        const options = {
-            versioning: bucketInfo.isVersioningEnabled(),
-            versionId,
-        };
+
+        let versionId = decodeVersionId(request.query);
+        let versioning = bucketInfo.isVersioningEnabled();
+
+        if (versionId === 'null') {
+            // Retrieve the null version id from the object metadata.
+            versionId = objMd && objMd.versionId;
+            if (!versionId) {
+                if (versioning) {
+                    // If the null version does not have a version id, it is a current null version.
+                    // To update the metadata of a current version, versioning is set to false.
+
+                    // This condition is to handle the case where a single null version looks like a master
+                    // key and will not have a duplicate versioned key and no version ID.
+                    // They are created when you have a non-versioned bucket with objects,
+                    // and then convert bucket to versioned.
+                    // If no new versioned objects are added for given object(s), they look like
+                    // standalone master keys.
+                    versioning = false;
+                } else {
+                    const versioningConf = bucketInfo.getVersioningConfiguration();
+                    // The purpose of this condition is to address situations in which
+                    // - versioning is "suspended" and
+                    // - no existing object or no null version.
+                    // In such scenarios, we generate a new null version and designate it as the master version.
+                    if (versioningConf && versioningConf.Status === 'Suspended') {
+                        versionId = '';
+                        omVal.isNull = true;
+                    }
+                }
+            }
+        }
```

Comment:
My comment might only be applicable for Zenko, but this doesn't work in versioning suspended mode.
When having a null version created after suspending the versioning, the options will equal `{ versionId: '$VERSION_ID' }` which will create another version instead of updating the master object.

Reply:
I'm unsure about which of the tests I wrote is incorrect. Please feel free to identify the test that is wrong.

Reply:
I just ran the tests from https://github.com/scality/cloudserver/pull/5171 and they all passed.

Reply:
Yes the tests pass as what i pointed out is not visible from the S3 API side.
Basically the object initially has only a master document in mongo and instead of only updating that, we create a new version object as well as update the mater.
I'm not entirely sure if this is an intended behaviour or not hence why i pointed it out.